.github/workflows/binaries-analysis.yml

```diff
@@ -0,0 +1,44 @@
+name: "Binaries Analysis"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '21 17 * * 0'
+
+jobs:
+  analyze:
+    name: BinSkim Binary Analyze
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        mode: [osx-x64,linux-x64,win-x64]
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+    - name: Setup NuGet.exe for use with actions
+      uses: NuGet/setup-nuget@v1.0.6
+    - name: Get BinSkim package from nuget
+      run: nuget install Microsoft.CodeAnalysis.BinSkim -Version 1.9.5 -OutputDirectory ${{ github.workspace }}
+    - name: Restore dependencies using Nuget
+      run: nuget restore src\client.sln -Verbosity Normal -NonInteractive -ConfigFile src/nuget.config
+    - name: dotnet Publish for ${{ matrix.mode }}
+      run: dotnet publish src\dsc\dsc.csproj -c Release -r ${{ matrix.mode }} --no-restore --self-contained true --verbosity normal
```

Comment:
☕️ we are using `dotnet` here, but I can't see the botnet action used before its use, how is this workflow getting the `dotnet`, might be we want something like this? https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-net#using-the-net-starter-workflow

Reply:
since I am using windows-latest runner, it comes with dot net pre-installed, please refer here - https://github.com/actions/runner-images/blob/main/images/win/Windows2022-Readme.md